src/mon/MDSMonitor.cc

```diff
@@ -646,6 +646,23 @@ bool MDSMonitor::prepare_beacon(MonOpRequestRef op)
 			m->get_name(), fsmap.get_epoch(), state, seq,
 			CEPH_FEATURES_SUPPORTED_DEFAULT));
     } else {
+      if (info.state == MDSMap::STATE_STANDBY && state != info.state) {
```

Comment:
Fair point, updated to flatten these conditions into the outer series of `else if`s